actors/util/adt/set.go

```diff
@@ -0,0 +1,74 @@
+package adt
+
+import (
+	"io"
+
+	cid "github.com/ipfs/go-cid"
+	hamt "github.com/ipfs/go-hamt-ipld"
+	cbg "github.com/whyrusleeping/cbor-gen"
+)
+
+var setValue hamtSetValue
+
+func init() {
+	setValue = hamtSetValue(1)
+}
+
+type hamtSetValue uint64
+
+func (s hamtSetValue) MarshalCBOR(w io.Writer) error {
+	if _, err := w.Write(cbg.CborEncodeMajorType(cbg.MajUnsignedInt, uint64(s))); err != nil {
+		return err
+	}
+	return nil
+}
+
+// Set stores data in a HAMT.
+type Set struct {
+	m *Map
+}
+
+// AsSet interprets a store as a HAMT-based set with root `r`.
+func AsSet(s Store, r cid.Cid) *Set {
+	return &Set{
+		m: AsMap(s, r),
+	}
+}
+
+// NewSet creates a new HAMT with root `r` and store `s`.
+func MakeEmptySet(s Store) (*Set, error) {
+	nd := hamt.NewNode(s)
+	newSet := AsSet(s, cid.Undef)
+	err := newSet.write(nd)
+	return newSet, err
```

Comment:
Use MakeEmptyMap and wrap the result rather than duplicate the internals. Remove the then-unused `write` implementation.